### PR TITLE
Config repo pipeline show read only configurations

### PR DESCRIPTION
**Pull request by GaneshSPatil**
#### Top Level Changes
  - [x] Show a warning indicating configuration can not be edited. Example: `Can not edit pipeline 'yaml-example' as it is defined in the config repository!`
  - [x] Show config repo information (config repo id, config repo url) where pipeline configuration is defined.
  - [x] Make `/go/admin/pipelines/yaml-example/general` page accessible for config repo pipelines
  - [x] Make dashboard pipeline gear/settings icon clickable for all the pipelines. Remove special handling of config repo pipelines.

#### Pipeline Config Edit Page Changes
  - [x] Disable/Remove `Save` button.
  - [x] Disable/Remove `Cancel` button.
  
  - [x] Pipeline
    - [x] General Options
    - [x] Project Management
    - [x] Stages
      - [x] Disable stage configuration type
      - [x] Disable removal of the stages
      - [x] Disable addition of new stage
      - [x] Allow clicking stage name link which can take a user to the stages tab
    - [x] Environment Variables
    - [x] Parameters

  - [x] Materials
    - [x] Disable removal of material.
    - [x] Disable adding of new material.
    - [x] Allow clicking on the material name to see the material configurations but do not allow modifying any material configurations.
  
  - [x] Stage
       - [x] Stage Settings
      - [x] Environment Variables
      - [x] Permissions
      - [x] Jobs
        - [x] Disable removal of the jobs
        - [x] Disable addition of a job
        - [x] Allow clicking the job name link which can take a user to the jobs tab

  - [x] Job
      - [x] Job Settings
      - [x] Artifacts
      - [x] Environment Variables
      - [x] Custom Tabs
      - [x] Tasks
          - [x] Disable removal of the task
        - [x] Disable addition of the task
        - [x] Allow clicking the task link to see the task configuration in a popup

  - [x] Task
      - [x] Show task configuration in read-only mode

  - [x] Templates
      - [x] Show templates read-only mode
      - [x] Show link to edit the template configuration
        
  - [ ] Fix pages linking to the pipeline config edit page. Tracked separately at: https://github.com/gocd/gocd/issues/7401

#### Screenshots

Screen Shot 2019-05-31 at 5 41 19 PM
Screen Shot 2019-05-31 at 5 41 34 PM
Screen Shot 2019-05-31 at 5 42 23 PM
Screen Shot 2019-05-31 at 5 42 31 PM
Screen Shot 2019-05-31 at 5 42 53 PM

### CODE REVIEW

**Comment by GaneshSPatil** [comment]
@arvindsv @maheshp @ketan 
This was a small spike to see if showing config repo pipeline configuration in read-only mode is possible or not. I think with some _more_ efforts and small code change its doable.
As we are starting work on the pipeline config spa, so not sure if it still makes sense to put in efforts on the older pages.

**Comment by arvindsv** [comment]
@GaneshSPatil Yes, given the work on pipeline config SPA, I think we should just hold on to this idea, to come back to if needed.

**Comment by GaneshSPatil** [comment]
@arvindsv @adityasood -- As we're not doing new pipeline config SPA, should we consider doing this?

@gocd/committers  Thoughts??

**Comment by arvindsv** [comment]
@GaneshSPatil I think it's worth a look.

File: server/config/cruise-config.xml [hunk 1]
```diff
@@ -47,4 +47,9 @@
       </stage>
     </pipeline>
   </pipelines>
+  <pipelines group="yaml-example" />
```
Comment (GaneshSPatil) [inline server/config/cruise-config.xml]: remove this??

**Comment by GaneshSPatil** [comment]
### Following things couldn't be tested (mentioned with reasons)
- **TFS materials**  (_Reason_: Defining TFS materials in the config repository is not supported by any of the config repository plugins). Issue: https://github.com/tomzo/gocd-yaml-config-plugin#tfs
- **Fetch External Artifacts Task** - (_Reason_: Defining fetch external artifact tasks has a bug in all config repository plugins.) Issue:  #7377  #7378

**Comment by GaneshSPatil** [comment]
#### Testing the changeset with YAML pipeline configuration:

##### prerequisite
1. Install `go-npm-poller` package repository plugin.
2. Define `npmjs` GoCD Package Repository.
3. Install `github-pr` pluggable scm plugin.
4. Install `docker-registry-artifact-plugin` external artifact plugin.
5. Define `store` the GoCD artifact store.
6. Define `up42` pipeline having `up42_stage` stage.
```yaml
format_version: 3
pipelines:
  pipeline1:
    group: first
    label_template: ${COUNT}
    lock_behavior: lockOnFailure
    tracking_tool:
      link: https://blah.com/${ID}
      regex: (+d)*
    display_order: -1
    timer:
      spec: 0 0 22 ? * *
      only_on_changes: true
    environment_variables:
      plain1: value1
      plain2: value2
    secure_variables:
      admin: AES:1PaiLQnpdzoknufcCdiJkA==:sI5UrvFmFJDvmN0NEzu8nQ==
    materials:
      git1:
        git: https://github.com/someone/someproject
        username: username
        shallow_clone: true
        auto_update: true
        branch: master
        destination: dest
        encrypted_password: AES:xdrx/ilIq0/v7gFwnLJHIA==:s4GuvBFK8UL/vTzU1IqzNw==
      pipeline-material:
        ignore_for_scheduling: true
        pipeline: up42
        stage: up42_stage
      subversion-1:
        svn: https://subversion.url.com
        username: admin
        check_externals: true
        auto_update: true
        destination: dest-2
        encrypted_password: AES:21RibmgqKr0PgZ46U3qngA==:/tX6KLB2t5RKJBSyfFGQcw==
      perforce:
        p4: https://perforceserver:567
        username: admin
        use_tickets: true
        view: this-is-a-very-pretty-perforce-view
        auto_update: true
        destination: dest-3
        encrypted_password: AES:9TANv/PaURebkL0ptKmzQQ==:1q/T4xYc8MZjTPdrBQndMw==
      mercurial:
        hg: https://mercurialserver.com
        username: admin
        auto_update: true
        branch: awesome
        destination: dest-4
        encrypted_password: AES:Xk7kcmDfGQ1lV6iVc3/T0g==:3/PY8d7OaWV29auuwiJwiQ==
      github-pr:
        scm: b787b203-091c-4fbf-881d-8b7d60b3f184
        blacklist:
        - blha/*
        destination: adsf-23
      npmjs_express:
        package: 513068fa-04b6-4093-82c1-1ea264eea07b
    stages:
    - stage:
        fetch_materials: true
        keep_artifacts: true
        clean_workspace: true
        approval:
          type: success
          allow_only_on_success: true
          roles:
          - xyz
          users:
          - jez
        environment_variables:
          env11: value11
        secure_variables:
          admin: AES:Sg3qhHAbAvgnEcFRLW0Csw==:d+e9SFOnrXn9ittPZRrX4Q==
        jobs:
          job:
            timeout: 10
            environment_variables:
              env1: value1
            secure_variables:
              admin: AES:/UmQCc+mLmct5zf9lp5jig==:XEaWlN6/Gmf1CLyqR27W/w==
            tabs:
              tab1: /path/to/home
              tab2: /path/to/home/again
            resources:
            - chrome
            - firefox
            - linux
            artifacts:
            - build:
                source: build-src
                destination: dest
            - test:
                source: test-src
                destination: testoutput
            - external:
                id: blah
                store_id: store
                configuration:
                  options:
                    BuildFile: ''
                    Tag: ''
                    Image: some-image
            tasks:
            - exec:
                on_cancel:
                  exec:
                    arguments:
                    - -f
                    - java
                    command: pkill
                    working_directory: something
                    run_if: passed
                arguments:
                - -alh
                command: ls
                working_directory: something
                run_if: failed
            - fetch:
                on_cancel:
                  exec:
                    arguments:
                    - -alh
                    command: ls
                    working_directory: something
                    run_if: passed
                is_file: true
                source: asfd
                destination: ''
                pipeline: up42
                stage: up42_stage
                job: up42_job
                run_if: passed
            - plugin:
                on_cancel:
                  exec:
                    arguments:
                    - -alhg
                    command: ls
                    working_directory: asdfa
                    run_if: passed
                configuration:
                  id: gradle.task.plugin
                  version: '1'
                options:
                  GradleHome: ''
                  Tasks: asdf
                  Daemon: 'true'
                  UseWrapper: 'true'
                  MakeWrapperExecutable: 'true'
                  Offline: 'true'
                  Debug: 'true'
                  AdditionalOptions: something:is-awesome
                run_if: passed
    parameters:
      param1: value1
      param2: value2
```

**Comment by GaneshSPatil** [comment]
Merging all the commits from this PR as is, as the commits are made considering individual page/section changes and also include some minor bug fixes. It will be easier to revert a particular commit in-case something goes wrong.

**Pull request merged**